File: program/print-diamond-pattern/print_diamond_pattern.go
package main

import "fmt"

func main() {
	var i, j, k int
	i = 1
	j = 1
	k = 1
	var row int
	row = 5
	for i = 1; i <= row; i++ {
		for j = 1; j <= row-i; j++ {
			fmt.Printf(" ")
		}
		for k = 1; k <= i*2-1; k++ {
			fmt.Printf("*")
		}
		fmt.Println()
	}
	for i = row - 1; i > 0; i-- {
		for j = 1; j <= row-i; j++ {
			fmt.Printf(" ")
		}
		for k = 1; k <= i*2-1; k++ {
			fmt.Printf("*")
		}
		fmt.Println()
	}
}

File: program/program/print-diamond-pattern/print_diamond_pattern.pl
# Perl program to print pattern 1

# Function to print pattern
sub print_pattern {
    my $num = shift;

    # Loop to iterate through each row
    for my $i (1..$num) {
        # Loop to print numbers in each row
        for my $j (1..$i) {
            print "$j ";
        }
        print "\n";
    }
}

# Main program
print "Enter the number of rows: ";
my $input = <STDIN>;
chomp($input);

# Check if input is a positive integer
if ($input =~ /^\d+$/ && $input > 0) {
    print_pattern($input);
} else {
    print "Invalid input. Please enter a positive integer.\n";
}
